src/lib/message/L2ToL1Message.ts

```diff
@@ -376,82 +365,88 @@ export class L2ToL1MessageReader extends L2ToL1Message {
   }
 
   /**
-   * Estimates the L1 block number in which this L2 to L1 tx will be available for execution
+   * Estimates the L1 block number in which this L2 to L1 tx will be available for execution.
+   * If the message can or already has been executed, this returns null
    * @param l2Provider
-   * @returns expected L1 block number where the L2 to L1 message will be executable
+   * @returns expected L1 block number where the L2 to L1 message will be executable. Returns null if the message can or already has been executed
    */
   public async getFirstExecutableBlock(
```

Comment:
maybe this should rly have estimate in the name? (don't rly care)

Reply:
yea true, `estimateExecutableBlock`